go/test/endtoend/reparent/newfeaturetest/reparent_in_tx_test.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2025 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package newfeaturetest
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/test/endtoend/reparent/utils"
+	"vitess.io/vitess/go/vt/vtctl/reparentutil/policy"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+var primary int
+
+// This test ensures that we get a VT15001 error when doing COMMIT while having primary being turned off.
+// The test then ensures that, if there was no partial commit, all the inserts were rolled back correctly.
+func testCommitError(t *testing.T, conn *mysql.Conn, clusterInstance *cluster.LocalProcessCluster, tablets []*cluster.Vttablet) {
+	tabletStopped := make(chan bool)
+	commitDone := make(chan bool)
+	idx := 1
+	createTxAndInsertRows(conn, t, &idx)
+
+	go func() {
+		<-tabletStopped
+		_, err := conn.ExecuteFetch("commit", 0, false)
+		require.ErrorContains(t, err, vterrors.VT15001(0).ID)
+		commitDone <- true
+	}()
+
+	reparent(t, clusterInstance, tablets, tabletStopped, commitDone)
+
+	_, err := conn.ExecuteFetch("delete from vt_insert_test", 0, false)
+	require.NoError(t, err)
```

Comment:
I think we should validate by selecting from the table that no data exists.

Reply:
Issue with selecting is that the commit can be a partial commit, leading to rows on the healthy shards.